web/src/app/lists/QueryList.js

```diff
@@ -133,8 +154,19 @@ QueryList.propTypes = {
   // If set, the search string param is ignored.
   noSearch: p.bool,
 
-  // filters additional to search, set in the search text field.
+  // controls unrelated to search, but still modify results, rendered to
+  // the left of the search text field
+  controls: p.node,
```

Comment:
Do we have more cases than check all/some + actions, or can we make the check all/none a feature of the list itself?

If we do that, we can stop tracking things like `onDataChange` (which duplicates data + double renders) and the indirection around it (incl. the redux action I think) can go away and instead just pass in something like menu options. Let's chat on this.

Reply:
We don't (check all, check none, ack, close, and escalate are all actions), but I'm not sure what this would look like without changing more of how it looks which we did want to avoid for now if we could help it

Reply:
I'd rather put the work into extending our re-usable components with features we use, than to have arbitrary "hook" points that we re-use in one-off places.

In this case, it would be useful to be able to do multi select on lists (e.g. remove multiple people from rotation, disable all integration keys temporarily, etc...) in the future, and get's us away from needing to do the bouncing of data around just to get it to work.

Reply:
I can play around with some concepts of the checkbox controls inside the card/list so the data is available without having to use redux, though it will more than likely delay getting this out and working on the next steps of decommissioning gql2